marlowe-dashboard-client/src/MainFrame/State.purs

```diff
@@ -33,22 +39,35 @@ mkMainFrame =
           }
     }
 
-handleQuery ::
-  forall a m.
-  Query a ->
-  HalogenM State Action ChildSlots Msg m (Maybe a)
+handleQuery :: forall a m. Query a -> HalogenM State Action ChildSlots Msg m (Maybe a)
 handleQuery (ReceiveWebSocketMessage msg next) = do
   current <- use _on
   modifying _on not
   pure $ Just next
 
-handleAction ::
-  forall m.
-  MonadAff m =>
-  Action ->
-  HalogenM State Action ChildSlots Msg m Unit
+handleAction :: forall m. MonadAff m => Action -> HalogenM State Action ChildSlots Msg m Unit
 handleAction Init = pure unit
 
+handleAction (ToggleOverlay overlay) = do
+  mCurrentOverlay <- use _overlay
+  case mCurrentOverlay of
+    Just currentOverlay -> assign _overlay $ if overlay == currentOverlay then Nothing else Just overlay
+    Nothing -> assign _overlay $ Just overlay
+
+handleAction (SetScreen screen) = do
+  assign _overlay Nothing
```

Comment:
Halogen re-draws the entire page per state modification, when they are multiple assigns is better to refactor to 

```purescript
modify_ (
  set _overlay Nothing 
  <<< set _card Nothing
  <<< set _screen screen
)
```

Reply:
Cool. Did not know about Halogen's `modify_`. Thanks. 👍